rotkehlchen/chain/ethereum/node_inquirer.py

```diff
@@ -281,17 +287,27 @@ def query_highest_block(self) -> BlockNumber:
 
         return BlockNumber(block_number)
 
-    def have_archive(self, requery: bool = False) -> bool:
-        if self.queried_archive_connection and requery is False:
-            return self.archive_connection
-
+    def determine_capabilities(self, web3: Web3) -> tuple[bool, bool]:
         balance = self.get_historical_balance(
             address=string_to_evm_address('0x50532e4Be195D1dE0c2E6DfA46D9ec0a4Fee6861'),
             block_number=87042,
+            web3=web3,
         )
-        self.archive_connection = balance is not None and balance == FVal('5.1063307')
-        self.queried_archive_connection = True
-        return self.archive_connection
+        is_archive = balance == FVal('5.1063307')
+
+        try:
+            tx = web3.eth.get_transaction(HexStr('0x5c504ed432cb51138bcf09aa5e8a410dd4a1e204ef84bfed1be16dfba1b22060'))  # noqa: E501
+            block_no = tx['blockNumber']
+        except (
+            requests.exceptions.RequestException,
+            TransactionNotFound,
+            BlockchainQueryError,
+            KeyError,
+        ):
+            block_no = None
+        is_pruned = block_no != 46147
```

Comment:
Why is there a need in checking block number? Won't it be sufficient to just check that the transaction was found?

Reply:
I am not so familiar with the nitty-gritty of ethereum but I guess you're right